android/src/main/java/com/expensify/livemarkdown/MarkdownRange.java

```diff
@@ -0,0 +1,38 @@
+package com.expensify.livemarkdown;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class MarkdownRange {
+  public final String type;
+  public final int start;
+  public final int end;
+  public final int length;
+  public final int depth;
+  public int index;
+
+  public MarkdownRange(String type, int start, int length, int depth, int index) {
+    this.type = type;
+    this.start = start;
+    this.length = length;
+    this.end = start + length;
+    this.depth = depth;
+    this.index = index;
+  }
+
+  public MarkdownRange(JSONObject range, int index) {
```

Comment:
It's kinda weird that `MarkdownRange` can be constructed only from `JSONObject` and there's no "basic" constructor – but it's a clever approach so I like it. Not sure if constructors are supposed to throw `RuntimeException`, maybe let's move the JSON part back to `parseRanges` and leave only "pure" logic here?

Reply:
Okay, fixed in https://github.com/Expensify/react-native-live-markdown/pull/534/commits/417ba57a49080096a0fea94a789a1909d2d3a5e4